### ISSUE TITLE
Pusher worker, always send 200 code status for `AuthError`

### ISSUE THREAD

**Issue by soufDev** [pusher/pusher-js#670]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

As we are implementing pusher with a `shared worker` we faced some issues that we think are really important to fix:
- when we receive an `authError` we receive always a `200` status due to the implementation in [here](https://github.com/pusher/pusher-js/blob/f00289c20e38b59f82902bd3fd7e60ec5108a810/src/runtimes/worker/auth/fetch_auth.ts#L47), so the quick fix I found is to send the status of the response in the error that we are throwing.
- we have an issue since we upgraded from `7.4.0` to `7.5.0` that is due to use of window in `randomInt` function in the worker code, so the fix is a simple one is to add `globalThis` and I think there is already a PR open #652. 

So The questions are:
- Do you think that we can create a fix for the status sent for the pusher worker ? if yes I can create a Pull request as soon as we are sure that this is the solution that we want to make.
- We need to move forward with #652 . is there any plan to merge it  sonner ? 

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/pusher-js/dist/worker/pusher.worker.js b/node_modules/pusher-js/dist/worker/pusher.worker.js
index e155eb2..ddab574 100644
--- a/node_modules/pusher-js/dist/worker/pusher.worker.js
+++ b/node_modules/pusher-js/dist/worker/pusher.worker.js
@@ -3639,7 +3639,7 @@ var fetchAuth = function (context, query, authOptions, authRequestType, callback
         if (status === 200) {
             return response.text();
         }
-        throw new HTTPAuthError(200, "Could not get " + authRequestType.toString() + " info from your auth endpoint, status: " + status);
+        throw new HTTPAuthError(status, "Could not get " + authRequestType.toString() + " info from your auth endpoint, status: " + status);
     })
         .then(function (data) {
         var parsedData;
@@ -3723,7 +3723,7 @@ var Worker = {
     },
     randomInt: function (max) {
         var random = function () {
-            var crypto = window.crypto || window['msCrypto'];
+            var crypto = globalThis.crypto || globalThis['msCrypto'];
             var random = crypto.getRandomValues(new Uint32Array(1))[0];
             return random / Math.pow(2, 32);
         };
diff --git a/node_modules/pusher-js/src/core/pusher.ts b/node_modules/pusher-js/src/core/pusher.ts
index e189d1d..1f470cd 100644
--- a/node_modules/pusher-js/src/core/pusher.ts
+++ b/node_modules/pusher-js/src/core/pusher.ts
@@ -10,7 +10,6 @@ import TimelineSender from './timeline/timeline_sender';
 import TimelineLevel from './timeline/level';
 import { defineTransport } from './strategies/strategy_builder';
 import ConnectionManager from './connection/connection_manager';
-import ConnectionManagerOptions from './connection/connection_manager_options';
 import { PeriodicTimer } from './utils/timers';
 import Defaults from './defaults';
 import * as DefaultConfig from './config';
```

**Comment by soufDev** [comment]
is there any update here ?

**Comment by soufDev** [comment]
here is the PR to fix this issue #681

**Comment by fbenevides** [comment]
Thanks for the contribution @soufDev.
